libraries/botbuilder-core/botbuilder/core/activity_handler.py

```diff
@@ -344,3 +364,58 @@ class ActivityHandler:
             conversation update, message reaction, or event activity, it calls this method.
         """
         return
+
+    async def on_invoke_activity(  # pylint: disable=unused-argument
+        self, turn_context: TurnContext
+    ) -> Union[InvokeResponse, None]:
+        """
+        Registers an activity event handler for the _invoke_ event, emitted for every incoming event activity.
+
+        :param turn_context: The context object for this turn
+        :type turn_context: :class:`botbuilder.core.TurnContext`
+
+        :returns: A task that represents the work queued to execute
+        """
+        try:
+            if (
+                turn_context.activity.name
+                == SignInConstants.verify_state_operation_name
+                or turn_context.activity.name
+                == SignInConstants.token_exchange_operation_name
+            ):
+                await self.on_sign_in_invoke(turn_context)
+                return self._create_invoke_response()
+
+            raise _InvokeResponseException(HTTPStatus.NOT_IMPLEMENTED)
```

Comment:
you have 502 behind the constant **HTTPStatus**.NOT_IMPLEMENTED but 200 behind **StatusCodes**.OK

that reads a little wonky - is there are reason for this?